spec/controllers/admin/posts_controller_spec.rb

```diff
@@ -0,0 +1,97 @@
+
+require 'rails_helper'
+
+describe Admin::PostsController do
+
+  before (:each) do
+    @user = FactoryGirl.create(:user)
+    @user.role = 'admin'
+    @user.save
+    sign_in @user
+  end
+
+  describe 'GET index' do
+    it 'assigns all admin_posts as @admin_posts' do
+      post = FactoryGirl.create(:post)
+      get :index
+      expect(assigns(:posts)).to eq([post])
+    end
+  end
+
+  describe 'GET show' do
+    it 'assigns the requested admin_post as @admin_post' do
+      post = FactoryGirl.create(:post)
+      get :show, {:id => post.to_param}
+      expect(assigns(:post)).to eq(post)
+    end
+  end
+
+  describe 'GET new' do
+    it 'assigns a new admin_post as @admin_post' do
+      get :new
+      expect(assigns(:post)).to be_a_new(Post)
+    end
+  end
+
+  describe 'GET edit' do
+    it 'assigns the requested admin_post as @admin_post' do
+      post = FactoryGirl.create(:post)
+      get :edit, {:id => post.to_param}
+      expect(assigns(:post)).to eq(post)
+    end
+  end
+
+  describe 'POST create' do
+    describe 'with valid params' do
+      it 'creates a new Admin::Post' do
+        expect {
+          post :create, post: FactoryGirl.attributes_for(:post)
+        }.to change(Post, :count).by(1)
+      end
+
+      it 'redirects to the created admin_post' do
+        post :create, post: FactoryGirl.attributes_for(:post)
+        expect(response).to redirect_to admin_posts_path
+      end
+    end
+  end
+
+  describe 'PUT update' do
+    describe 'with valid params' do
+
+      it 'updates the requested admin_post' do
+        post = FactoryGirl.create(:post)
+        put :update, id: post.to_param, post: FactoryGirl.attributes_for(:post)
+        post.reload
+      end
+
+      it 'assigns the requested admin_post as @admin_post' do
+        post = FactoryGirl.create(:post)
+        put :update, id: post.to_param, post: FactoryGirl.attributes_for(:post)
+        expect(assigns(:post)).to eq(post)
+      end
+
+      it 'redirects to the admin_post' do
+        post = FactoryGirl.create(:post)
+        put :update, id: post.to_param, post: FactoryGirl.attributes_for(:post)
+        expect(response).to redirect_to admin_post_path
+      end
+    end
+  end
+
+  describe 'DELETE destroy' do
+
+    it 'destroys the requested admin_post' do
+      post = FactoryGirl.create(:post)
+      expect {
+        delete :destroy, id: post.to_param
+      }.to change(Post, :count).by(-1)
+  end
```

Comment:
`end` at 89, 2 is not aligned with `it 'destroys the requested admin_post' do` at 84, 4